src/java.xml/share/classes/com/sun/org/apache/xalan/internal/utils/XMLSecurityManager.java

```diff
@@ -259,14 +259,17 @@ public void setLimit(int index, State state, Object value) {
             //if it's explicitly set, it's treated as yes no matter the value
             printEntityCountInfo = (String)value;
         } else {
-            int temp = 0;
-            try {
+            int temp;
+            if (Integer.class.isAssignableFrom(value.getClass())) {
+                temp = (Integer)value;
```

Comment:
Why not use pattern matching?
`if (value instanceof Integer intValue) {
  temp = intValue;
}

Reply:
Good question. Given that the earlier changes in the whole set (impl-specific properties) were backported, I assumed this would too, to be complete. Code consistency and clean backport may exceed the benefit of a more advanced code feature (e.g. pattern matching). 

I also am still doing works based on JDK 8, that provides a much quicker turnaround.

Saw Patrick's pattern matching changesets for some lib code. It's possible we could do it as well for the xml code. But in general, xml code, esp. those from Apache support a lower source level.

Reply:
ok, JDK 8 could use instanceof instead of isAssignable.

Reply:
Thanks. Updated.